code/processes/segmentedtickerplant.q

```diff
@@ -83,5 +83,38 @@ init:{
   .stplg.init[string .proc.procname];
  };
 
-// Have the init function called from torq.q
+// Loads the striping.json config file and logs any problems
+configload:{[scpath]
+     @[jsonchecks; 
+       .stpps.stripeconfig:@[{.j.k read1 x};scpath;{.lg.o[`configload;"Failed to load in striping.json file: ",x]}];
+       {.lg.o[`configload;"Datastriping is not enabled due to problems with striping.json"]}]
+     };
+
+// Checks if each subscriptiondefault is set for each segment and errors if not defined
+jsonchecks:{[.stpps.stripeconfig]
+     // check defaults are ignore or all
+     defaults:{first (flip .stpps.stripeconfig[x])[`subscriptiondefault]}each key .stpps.stripeconfig;
+     errors:1+ where {[x] not ("ignore"~x) or ("all"~x)}each defaults;
+     {if[0<count x;.lg.o[`jsonchecks;m:"subscriptiondefault not defined as \"ignore\" or \"all\" for segment ",string[x]," "]]}each errors;
+     // check for valid tables
+     keydict: key(flip .stpps.stripeconfig[`segid]);
```

Comment:
`keydict` isn't a very descriptive varibale name - maybe something like `configtables` (I'm assuming from the next few lines that this variable contains the tables that are listed in the config file?)